packages/shared/src/components/notifications/ReminderModal.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { FormEventHandler, ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { format } from 'date-fns';
+import { Modal, ModalProps } from '../modals/common/Modal';
+import { ModalHeader } from '../modals/common/ModalHeader';
+import { ModalBody } from '../modals/common/ModalBody';
+import { ModalSize } from '../modals/common/types';
+import { Button, ButtonVariant } from '../buttons/Button';
+import {
+  getRemindAt,
+  ReminderPreference,
+} from '../../hooks/notifications/useBookmarkReminder';
+
+export type ReminderModalProps = {
+  onReminderSet: (reminder: string) => void;
+} & Pick<ModalProps, 'isOpen' | 'onRequestClose'>;
```

Comment:
Most of the time, we prefer interface over type intersections. More information here: https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections

In summary, extending interfaces is more effective, cached, and displays conflicts.

Reply:
Cool, thanks for sharing!